cloudify_cli/commands/install.py

```diff
@@ -0,0 +1,142 @@
+
+    # First, make sure the `blueprint-path` wasn't supplied with
+    # `archive_location` or with `blueprint_filename`
+    check_for_mutually_exclusive_arguments(blueprint_path,
+                                           archive_location,
+                                           blueprint_filename)
+
+    # The presence of the `archive_location` argument is used to distinguish
+    # between `install` in 'blueprints upload' mode,
+    # and `install` in 'blueprints publish archive' mode.
+    if archive_location:
+
+        if blueprint_filename is None:
+            blueprint_filename = DEFAULT_BLUEPRINT_FILE_NAME
+
+        # If blueprint-id wasn't supplied, assign it to the name of the archive
+        if blueprint_id is None:
+
+            blueprints.check_if_archive_type_is_supported(archive_location)
+
+            (archive_location, archive_location_type) = \
+                blueprints.determine_archive_type(archive_location)
+
+            # if the archive is a local path, assign blueprint-id the name of
+            # the archive file without the extension
+            if archive_location_type == 'path':
+                filename, ext = os.path.splitext(
+                        os.path.basename(archive_location))
+                blueprint_id = filename
+
+            # if the archive is a url, assign blueprint-id name of the file
+            # that the url leads to, without the extension.
+            # e.g. http://example.com/path/archive.zip?para=val#sect -> archive
+            if archive_location_type == 'url':
+
+                path = urlparse.urlparse(archive_location).path
+                archive_file = path.split('/')[-1]
+                archive_name = archive_file.split('.')[0]
+                blueprint_id = archive_name
+
+        blueprints.publish_archive(archive_location, blueprint_filename,
+                                   blueprint_id)
+    else:
+
+        if blueprint_path is None:
+            blueprint_path = os.path.join(utils.get_cwd(),
+                                          DEFAULT_BLUEPRINT_PATH)
+
+        # If blueprint-id wasn't supplied, assign it to the name of
+        # folder containing the application's blueprint file.
+        if blueprint_id is None:
+
+            blueprint_id = os.path.basename(
+                    os.path.dirname(
+                            os.path.abspath(
+                                    blueprint_path)))
+        # Try opening `blueprint_path`, since `blueprints.upload` expects the
+        # `blueprint_path` argument to be a file.
+        # (The reason for this is beyond me. That's just the way it is)
+        try:
+            with open(blueprint_path) as blueprint_file:
+                blueprints.upload(blueprint_file,
+                                  blueprint_id,
+                                  validate_blueprint)
+        except IOError:
+            raise CloudifyCliError("Can't open the the file that "
```

Comment:
1. Add the blueprint_path in the error message.
2. delete the parenthesis
3. Add information about the original IOError (e.g show it in parenthesis).